ethcore/sync/src/block_sync.rs

```diff
@@ -515,6 +528,7 @@ impl BlockDownloader {
 				},
 				Err(BlockImportError(BlockImportErrorKind::Queue(QueueErrorKind::Full(limit)), _)) => {
 					debug!(target: "sync", "Block import queue full ({}), restarting sync", limit);
+					bad = true;
```

Comment:
This is done to trigger resetting the sync right? Do you think we should return a different error type other than `BlockDownloaderImportError::Invalid`? (And also handle it in `ChainSync::collect_blocks`).

Reply:
Yes I agree and I had previously done that. However my idea with this initial PR was to do the minimum amount to fix this bug.

However looks like this fix doesn't work entirely so I will add that back while waiting for another mainnet sync...